xbmc/platform/android/activity/JNIXBMCRecognitionListener.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "JNIXBMCRecognitionListener.h"
+
+#include "CompileInfo.h"
+#include "dialogs/GUIDialogKaiToast.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindow.h"
+#include "guilib/GUIWindowManager.h"
+#include "messaging/ApplicationMessenger.h"
+#include "ServiceBroker.h"
+#include "utils/log.h"
+
+#include <androidjni/Context.h>
+#include <androidjni/jutils-details.hpp>
+
+using namespace jni;
+
+static std::string s_className = std::string(CCompileInfo::GetClass()) + "/interfaces/XBMCRecognitionListener";
+
+CJNIXBMCRecognitionListener::CJNIXBMCRecognitionListener() : CJNIBase(s_className)
+{
+  m_object = new_object(CJNIContext::getClassLoader().loadClass(GetDotClassName(s_className)));
+  m_object.setGlobal();
+
+  add_instance(m_object, this);
+}
+
+void CJNIXBMCRecognitionListener::RegisterNatives(JNIEnv* env)
+{
+  jclass cClass = env->FindClass(s_className.c_str());
+  if(cClass)
+  {
+    JNINativeMethod methods[] =
+    {
+      {"_onReadyForSpeech", "(Landroid/os/Bundle;)V", (void*)&CJNIXBMCRecognitionListener::_onReadyForSpeech},
+      {"_onError", "(I)V", (void*)&CJNIXBMCRecognitionListener::_onError},
+      {"_onResults", "(Landroid/os/Bundle;)V", (void*)&CJNIXBMCRecognitionListener::_onResults},
+    };
+
+    env->RegisterNatives(cClass, methods, sizeof(methods)/sizeof(methods[0]));
+  }
+}
+
+CJNIXBMCRecognitionListener::~CJNIXBMCRecognitionListener()
+{
+  remove_instance(this);
+}
+
+void CJNIXBMCRecognitionListener::_onReadyForSpeech(JNIEnv* env, jobject thiz, jobject bundle)
+{
+  CLog::Log(LOGDEBUG, "CJNIXBMCRecognitionListener onReadyForSpeech()");
+  CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Info, "Speech to text", "Listening...");
+}
+
+void CJNIXBMCRecognitionListener::_onError(JNIEnv* env, jobject thiz, jint i)
+{
+  std::string error = "Error";
+
+  if (i == CJNISpeechRecognizer::ERROR_AUDIO)
+    error = "Audio recording error";
+  else if (i == CJNISpeechRecognizer::ERROR_CLIENT)
+    error = "Client side errors";
+  else if (i == CJNISpeechRecognizer::ERROR_INSUFFICIENT_PERMISSIONS)
+    error = "Insufficient permissions";
+  else if (i == CJNISpeechRecognizer::ERROR_NETWORK)
+    error = "Network related errors";
+  else if (i == CJNISpeechRecognizer::ERROR_NETWORK_TIMEOUT)
+    error = "Network operation timed out";
+  else if (i == CJNISpeechRecognizer::ERROR_NO_MATCH)
+    error = "No recognition result matched";
+  else if (i == CJNISpeechRecognizer::ERROR_RECOGNIZER_BUSY)
+   error = "RecognitionService busy";
+  else if (i == CJNISpeechRecognizer::ERROR_SERVER)
+    error = "Server sends error status";
+  else if (i == CJNISpeechRecognizer::ERROR_SPEECH_TIMEOUT)
+    error = "No speech input";
```

Comment:
replace if / else / else if with switch / case

Those error strings should be localized, not hardcoded in English.

Reply:
Not sure these errors are relevant for the end user. should those just be logged instead of confronting the user with technical stuff they do not understand?